### PR TITLE
Use locale when sorting autogenerated sidebar entries

### PR DESCRIPTION
**Pull request by HiDeoo**
#### What kind of changes does this PR include?

- Changes to Starlight code

#### Description

This pull request adds support for using the locale when sorting autogenerated sidebar entries.

At the moment, the current sorting algorithm, ignoring pages with an explicit order which is an early return, is as follows:

```ts
// If two pages have the same order value they will be sorted by their slug.
return a.slug < b.slug ? -1 : a.slug > b.slug ? 1 : 0;
```

If, for example, we have files with the slugs `route`, `radio` and `référence`, the following order will be used:

- `radio`
- `route`
- `référence`

Unfortunately, this is not the expected order for French users for example, who would expect the following order:

- `radio`
- `référence`
- `route`

When sorting directories, I am not sure we are actually doing any sorting at all as the `slug` would be `undefined` and having `undefined` in one of the comparisons will always return `0` and thus not changing the order.

To change this behavior, this PR uses an [`Intl.Collator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator) (Node.js v13.0.0) to do language-sensitive string comparison.

### CODE REVIEW

**Review by dreyfus92 — APPROVED** [review]
This is a nice catch 🤯 @HiDeoo, LGTM ✅

File: packages/starlight/utils/navigation.ts [hunk 1]
```diff
@@ -184,13 +184,17 @@ function getOrder(routeOrDir: Route | Dir): number {
 }
 
 /** Sort a directory’s entries by user-specified order or alphabetically if no order specified. */
-function sortDirEntries(dir: [string, Dir | Route][]): [string, Dir | Route][] {
-	return dir.sort(([, a], [, b]) => {
+function sortDirEntries(
+	dir: [string, Dir | Route][],
+	locale: string | undefined
+): [string, Dir | Route][] {
+	const collator = new Intl.Collator(localeToLang(locale));
```
Comment (delucis) [inline packages/starlight/utils/navigation.ts:191]: Really nice! I didn’t know about `Intl.Collator` — so nice how much i18n stuff like this is built into Node these days.

**Review by delucis — APPROVED** [review]
This looks great! Thank you @HiDeoo 💖

File: packages/starlight/utils/navigation.ts (hunk 1 above)
Comment (HiDeoo) [reply packages/starlight/utils/navigation.ts:191]: Definitely, and super handy. I think that is what is used under the hood by the String [`localeCompare()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) but they recommend creating an `Intl.Collator` object for repeated use.

Just to double-check for sanity, I can safely merge this one as it is a patch and it does not update the docs, right?

Comment (delucis) [reply packages/starlight/utils/navigation.ts:191]: > I can safely merge this one as it is a patch and it does not update the docs, right?

Yes! Safe to merge.

**Pull request merged**